native/markdown/rules.react.js

```diff
@@ -49,61 +57,45 @@ export default function rules(styles: StyleSheetOf<MarkdownStyles>) {
       ...SimpleMarkdown.defaultRules.paragraph,
       // simple-markdown collapses multiple newlines into one, but we want to
       // preserve the newlines
-      match: SimpleMarkdown.blockRegex(/^([^\n]*)(?:\n|$)/),
+      match: SimpleMarkdown.blockRegex(/^((?:[^\n]*)(?:\n|$))/),
```

Comment:
Note that the capture group (the outer parentheses) now includes the newline character